cmake/GenerateVersionHeader.cmake

```diff
@@ -0,0 +1,41 @@
+# Split version string into components
+string(REGEX MATCH "^([0-9]+)\\.([0-9]+)\\.([0-9]+)" _ ${CELERITY_VERSION})
+set(CELERITY_VERSION_MAJOR ${CMAKE_MATCH_1})
+set(CELERITY_VERSION_MINOR ${CMAKE_MATCH_2})
+set(CELERITY_VERSION_PATCH ${CMAKE_MATCH_3})
+
+message(VERBOSE "Celerity version is ${CELERITY_VERSION_MAJOR}.${CELERITY_VERSION_MINOR}.${CELERITY_VERSION_PATCH}")
+
+# Attempt to obtain git revision / dirty status
+set(CELERITY_VERSION_GIT_REVISION "unknown")
+set(CELERITY_VERSION_GIT_IS_DIRTY 0)
+
+find_package(Git QUIET)
+if(GIT_FOUND)
+  execute_process(
+    COMMAND ${GIT_EXECUTABLE} rev-parse --short HEAD
+    WORKING_DIRECTORY "${CELERITY_SOURCE_DIR}"
+    RESULT_VARIABLE EXIT_CODE
+    OUTPUT_VARIABLE GIT_REVISION
+    ERROR_QUIET
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+  )
+  if(EXIT_CODE EQUAL 0)
+    set(CELERITY_VERSION_GIT_REVISION ${GIT_REVISION})
+    execute_process(
+      COMMAND ${GIT_EXECUTABLE} diff-index --quiet HEAD --
+      WORKING_DIRECTORY "${CELERITY_SOURCE_DIR}"
+      RESULT_VARIABLE CELERITY_VERSION_GIT_IS_DIRTY
+      ERROR_QUIET
+      OUTPUT_STRIP_TRAILING_WHITESPACE
```

Comment:
The output is not captured here, so this flag is superfluous.

Reply:
Fixed